src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -1697,6 +1701,7 @@ - (void)_handleReportBegin
     assertChipStackLockedByCurrentThread();
 
     std::lock_guard lock(_lock);
+    self.lastSubscriptionActiveTime = [NSDate now];
```

Comment:
Why is this here (where injections will trigger it), not in the subscription callbacks like the other callsites?

Reply:
Given this is called directly and already locks, I thought it's best to keep all locked operations in the same block. I can move it if you feel I should.

Reply:
Moved.